$NetBSD: patch-include_llvm_Analysis_ConstantFolding.h,v 1.1 2018/12/15 23:41:45 jperkin Exp $

Fix SunOS CS conflict not handled by include/llvm-c/DataTypes.h

diff -wpruN '--exclude=*.orig' a~/include/llvm/Analysis/ConstantFolding.h a/include/llvm/Analysis/ConstantFolding.h
--- a~/include/llvm/Analysis/ConstantFolding.h	1970-01-01 00:00:00
+++ a/include/llvm/Analysis/ConstantFolding.h	1970-01-01 00:00:00
@@ -19,6 +19,10 @@
 #ifndef LLVM_ANALYSIS_CONSTANTFOLDING_H
 #define LLVM_ANALYSIS_CONSTANTFOLDING_H
 
+#ifdef __sun
+#undef CS
+#endif
+
 namespace llvm {
 class APInt;
 template <typename T> class ArrayRef;
